src/parser/shared/normalizers/Channeling.ts

```diff
@@ -0,0 +1,344 @@
+    // Mage
+    buffChannelSpec(SPELLS.EVOCATION.id),
+    buffChannelSpec(SPELLS.SHIFTING_POWER.id),
+    nextCastChannelSpec(SPELLS.ARCANE_MISSILES.id),
+    // Warlock
+    buffChannelSpec(SPELLS.DRAIN_SOUL_TALENT.id),
+    // Priest
+    buffChannelSpec(SPELLS.DIVINE_HYMN_CAST.id),
+    nextCastChannelSpec(SPELLS.PENANCE_CAST.id),
+    buffChannelSpec(SPELLS.VOID_TORRENT_TALENT.id),
+    buffChannelSpec(SPELLS.MIND_FLAY.id), // TODO double check ID
+    buffChannelSpec(SPELLS.MIND_SEAR.id), // TODO double check ID
+    // Rogue
+    // Druid
+    buffChannelSpec(SPELLS.CONVOKE_SPIRITS.id),
+    // Monk
+    buffChannelSpec(SPELLS.ZEN_MEDITATION.id),
+    buffChannelSpec(SPELLS.ESSENCE_FONT.id),
+    buffChannelSpec(SPELLS.SOOTHING_MIST.id),
+    buffChannelSpec(SPELLS.CRACKLING_JADE_LIGHTNING.id),
+    buffChannelSpec(SPELLS.FISTS_OF_FURY_CAST.id),
+    // Demon Hunter
+    buffChannelSpec(SPELLS.EYE_BEAM.id), // TODO special handling because of the two buffs?
+    // Shaman
+    // Hunter
+    buffChannelSpec(SPELLS.RAPID_FIRE.id),
+    // TODO Barrage - can't find a single log with someone taking the talent...
+    // Paladin
+    // Warrior
+    buffChannelSpec(SPELLS.BLADESTORM_TALENT.id),
+    buffChannelSpec(SPELLS.BLADESTORM.id),
+    // Death Knight
+    buffChannelSpec(SPELLS.BLOODDRINKER_TALENT.id),
+  ];
+
+  // registered special case handlers, mapped by guid
+  channelSpecMap: { [key: number]: ChannelHandler } = {};
+
+  /**
+   * Constructs a Channeling normalizer that deals with normal hardcasts and instants by default,
+   * and then allows special case functions to be registered by spellId for channels.
+   * @param channelSpecs special case functions that handle specific spells
+   */
+  constructor(options: Options) {
+    super(options);
+
+    // populate the specs into a mapping for quick handling
+    Channeling.CHANNEL_SPECS.forEach((spec) => {
+      spec.guids.forEach((guid) => {
+        if (this.channelSpecMap[guid]) {
+          console.error(
+            'Channeling module trying to add more than one handler for spell guid ' +
+              guid +
+              ' - only the latest handler will apply!',
+          );
+        }
+        this.channelSpecMap[guid] = spec.handler;
+      });
+    });
+  }
+
+  normalize(events: AnyEvent[]) {
+    const eventsInserter: InsertableEventsWrapper = new InsertableEventsWrapper(events);
+
+    const channelState: ChannelState = {
+      unresolvedChannel: null,
+      eventsInserter,
+    };
+
+    // for each event, check if there is a special case handler for that GUID -
+    // if so - call the handler, if not - call the default handling
+    events.forEach((event: AnyEvent, index: number) => {
+      if (!HasSource(event) || event.sourceID !== this.selectedCombatant.id) {
```

Comment:
If this does plan to work with pets you will need to factor in another field
I believe we don't currently map it but it should be sourceInstance
which would be which instance of the pet it is. This would/could be something major for Venthyr Mistweaver and Demo lock 

You can actually see this field being used in other locations in the codebase...mainly where summoning/tracking pets are concerned

Reply:
Yeah if I can be convinced it's valuable to support pets, I can see where I would extend this.

Reply:
i'm not aware with any pets with channels except succubus CC, but that doesn't impact downtime calculations so i think its fine.

Reply:
I'm gonna say this should go live without pet support, and it can be added later if needed.